### PR TITLE
publishers: Add Zulip publisher.

### PR DESCRIPTION
**Pull request by fishinthecalculator**
I added a `_send_private` function as well: it just sends a DM to a given user. You can test it by sending a message to yourself via the web gui, noting down the integer in the URL (it'll be something like https://zulip.twc-italia.org/#narrow/pm-with/666-user666,  where 666 is your user id), and then just paste it in the function.

### CODE REVIEW

**Comment by tinoilcotechino** [comment]
should fix #57

File: mobilizon_reshare/publishers/zulip.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    _conf = ("publisher", "zulip")
+    default_template_path = pkg_resources.resource_filename(
+        "mobilizon_reshare.publishers.templates", "zulip.tmpl.j2"
+    )
+    api_uri = "https://zulip.twc-italia.org/api/v1/"
+
+    def _send_private(self, message: str) -> Response:
+        """
+        # For private messages
```
Comment (chobeat) [inline mobilizon_reshare/publishers/zulip.py:29]: I think this comment is not needed

File: mobilizon_reshare/publishers/zulip.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                "type": "private",
+                "to": f"[{self.user_id}]",
+                "content": message,
+            },
+        )
+
+    def _send(self, message: str) -> Response:
+        """
+        # For stream messages
+        curl -X POST https://yourZulipDomain.zulipchat.com/api/v1/messages \
```
Comment (chobeat) [inline mobilizon_reshare/publishers/zulip.py:50]: same

File: mobilizon_reshare/publishers/zulip.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                raise_error=InvalidResponse,
+            )
+
+        return data
+
+    def validate_message(self) -> None:
+        # TODO implement
+        pass
+
+    def _preprocess_event(self):
```
Comment (chobeat) [inline mobilizon_reshare/publishers/zulip.py:140]: I think this is not used here, it's a copypaste

**Review by chobeat — CHANGES_REQUESTED** [review]
Also add a couple tests mocking the zulip response

**Comment by tinoilcotechino** [comment]
whoops i forgot the test, ignore the reviewing request

**Comment by chobeat** [comment]
is it done?

**Comment by tinoilcotechino** [comment]
It's finished. I threw in some small changes to the release script

File: tests/publishers/conftest.py [hunk 4]
```diff
@@ -2,15 +2,36 @@
 from uuid import UUID
 
 import pytest
+import responses
 
 from mobilizon_reshare.event.event import MobilizonEvent
 from mobilizon_reshare.publishers.abstract import AbstractPublisher
 from mobilizon_reshare.publishers.exceptions import PublisherError, InvalidResponse
 
+api_uri = "https://zulip.twc-italia.org/api/v1/"
```
Comment (chobeat) [inline tests/publishers/conftest.py:11]: all this stuff is specific to zulip so it shouldn't be in the conftest for all the publishers, especially if it has generic names. Move it to test_zulip or create a folder for all the zulip stuff

**Review by chobeat — CHANGES_REQUESTED** [review]

**Pull request merged**